[video_player_avfoundation] Fix playback speed doesn't get persisted on iOS after reinitializing with new VideoPlayerController

vmalyi opened the pull request:
This PR fixes the https://github.com/flutter/flutter/issues/100351 by using the `playImmediately(atRate:)` instead of `play()` because calling `play()` has an effect of setting back the `rate` to 1.0 which is unwanted in case of `VideoPlayerController`'s reinitialization.

Using `playImmediately(atRate:)` is safe even with slower internet connections because AVPlayer automatically compensates eventual lack of media in the buffer by simply behaving as if it has encountered a stall during playback. 

Considering that `updatePlayingState()` function which, in turn, invokes the `playImmediately(atRate:)`, is only called when handling "pause" or "play" events or in `AVPlayerItemStatusReadyToPlay` state, it's safe for user experience to use the `playImmediately(atRate:)` instead of `play()`.


## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

jmagman commented on packages/video_player/video_player_avfoundation/CHANGELOG.md:
```diff
@@ -1,3 +1,7 @@
+## 2.4.3
+
+* Fixes playback speed doesn't get persisted after reinitializing with new VideoPlayerController (https://github.com/flutter/flutter/issues/100351)
```
We don't usually link to the GitHub issue.
```suggestion
* Fixes playback speed doesn't get persisted after reinitializing with new VideoPlayerController.
```

jmagman reviewed (changes requested):
This will need a test.  Check out https://github.com/flutter/packages/blob/main/packages/video_player/video_player_avfoundation/test/avfoundation_video_player_test.dart, hopefully you can reproduce the failure there, and prove your PR fixes it.

stuartmorgan commented:
@vmalyi Are you still planning on updating this PR based on the feedback above?

vmalyi commented:
> @vmalyi Are you still planning on updating this PR based on the feedback above?

@stuartmorgan, sorry for delay here. 

Yeah, this will be ready for another review this week.

vmalyi commented:
@stuartmorgan, sorry for inconvenience: I've accidentally closed this PR and couldn't reopen it. 

The new PR addressing all comments is now: https://github.com/flutter/packages/pull/3720
